pkg/liquidity-mining/contracts/gauges/StakelessGaugeController.sol

```diff
@@ -0,0 +1,223 @@
+        uint256 totalGauges = _gauges[IGaugeAdder.GaugeType.Arbitrum].length();
+        if (totalGauges == 0) {
+            return (0, 0);
+        }
+        EnumerableSet.AddressSet storage arbitrumGauges = _gauges[IGaugeAdder.GaugeType.Arbitrum];
+
+        // We can do this cast safely because we check that the gauge was created from the ArbitrumRootGaugeFactory
+        // when we added it to the address set.
+        uint256 singleGaugeBridgeCost = ArbitrumRootGauge(arbitrumGauges.unchecked_at(0)).getTotalBridgeCost();
+        uint256 totalArbETH = singleGaugeBridgeCost * arbitrumGauges.length();
+        require(msg.value >= totalArbETH, "Not enough eth to cover arbitrum checkpoints");
+
+        return (singleGaugeBridgeCost, totalArbETH);
+    }
+
+    /**
+     * @dev Performs checkpoints for all gauges of the given type whose relative weight is at least the specified one.
+     * @param gaugeType - Type of the gauges to checkpoint.
+     * @param minRelativeWeight - Threshold to filter out gauges below it.
+     * @param currentPeriod - Current block time rounded down to the start of the week.
+     * @param costPerCheckpoint - Value in ETH to be spent for each gauge to checkpoint to cover bridging costs.
+     * This method doesn't check whether the caller transferred enough ETH to cover the whole operation.
+     */
+    function _checkpointGauges(
+        IGaugeAdder.GaugeType gaugeType,
+        uint256 minRelativeWeight,
+        uint256 currentPeriod,
+        uint256 costPerCheckpoint
+    ) private {
+        uint256 totalGauges = _gauges[gaugeType].length();
+        EnumerableSet.AddressSet storage gaugeAddressSet = _gauges[gaugeType];
+        for (uint256 i = 0; i < totalGauges; ++i) {
+            address gauge = gaugeAddressSet.unchecked_at(i);
+            // Skip gauges that are below the threshold.
+            if (_gaugeController.gauge_relative_weight(gauge, currentPeriod) < minRelativeWeight) {
+                continue;
+            }
+
+            _authorizerAdaptor.performAction{ value: costPerCheckpoint }(
+                gauge,
+                abi.encodeWithSelector(IStakelessGauge.checkpoint.selector)
+            );
+        }
+    }
+
+    /**
+     * @dev Rounds the provided timestamp down to the beginning of the current week (Thurs 00:00 UTC).
+     */
+    function _roundDownTimestamp(uint256 timestamp) private pure returns (uint256) {
+        // Division by zero or overflows are impossible here.
+        return (timestamp / 1 weeks) * 1 weeks;
+    }
+
+    /**
+     * @dev Returns true if gauge type is Polygon, Arbitrum, Optimism, Gnosis or ZKSync; false otherwise.
+     */
+    function _isValidGaugeType(IGaugeAdder.GaugeType gaugeType) private pure returns (bool) {
+        return
+            uint8(gaugeType) >= uint8(IGaugeAdder.GaugeType.Polygon) &&
+            uint8(gaugeType) <= uint8(IGaugeAdder.GaugeType.ZKSync);
```

Comment:
:scream:

A big `type == enum.Value || ...` chain would be much more explicit :grin: The gas savings this would introduce are minimal compared to the rest of the execution.

Reply:
done